tests/Integration/DomainApiTest.php

```diff
@@ -115,7 +116,7 @@ public function testDomainCreate_DuplicateValues()
             false                       // wildcard domain?
         );
         $this->assertNotNull($domain);
-        $this->assertInstanceOf(CreateResponse::class, $domain);
+        $this->assertInstanceOf(ErrorResponse::class, $domain);
```

Comment:
There is a lot of changes not related to the "deserialize" issue. Try to put these changes in a separate PR in the future. It's okey for now.